hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -138,10 +137,16 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
   override protected def getPartitions: Array[Partition] =
     fileSplits.zipWithIndex.map(file => HoodieMergeOnReadPartition(file._2, file._1)).toArray
 
-  private def getConfig: Configuration = {
-    val conf = confBroadcast.value.value
-    CONFIG_INSTANTIATION_LOCK.synchronized {
-      new Configuration(conf)
-    }
+  private def getHadoopConf: Configuration = {
+    val conf = hadoopConfBroadcast.value.value
+    new Configuration(conf)
```

Comment:
still need the lock for concurrency?

Reply:
I don't actually think we need the lock

Reply:
But even more importantly -- there's no real concurrent access to this method

Reply:
This lock is from the beginning of the implementation of `HoodieMergeOnReadRDD` and it is likely due to the broadcast configuration issue in Spark 2.4 (https://github.com/apache/hudi/pull/1848#issuecomment-666091024).  @garyli1019 could you confirm if that's the case?  I'd rather keep it and keep the code safe from potentially breaking the MOR queries.

Reply:
How would it break MOR queries though?

See whichever way we turn this around this lock doesn't really make sense:
 - There's no concurrency accessing this method, each task gets its own copy of the object
 - Only concurrency is w/in accessing the Broadcast shared cache which is hedged by its own lock

Reply:
Synced up offline.  @alexeykudinkin and I are aligned in that we need to clean up the legacy code and remove any unnecessary code path.  For this particular case, there could be a problem regarding the modification of the hadoop conf returned by this function (check: `HoodieMergeOnReadRDD#compute` -> `LogFileIterator.logRecords` -> `scanLog` -> `FSUtils.getFs` -> `prepareHadoopConf` -> `conf.set`).   That's likely why the lock is put in from the beginning.  So the change is going to be reverted in this PR and we'll revisit this in a separate PR to be merged after 0.13.0.

Reply:
@yihua Hi Ethan, the hadoop configuration issue that I can recall was related to serialization during the broadcast. The hadoop configuration was not serializable. The initial implementation of `HoodieMergeOnReadRDD` somehow new a Configuration object or use a serializableConfiguration to solve this issue.